### PR TITLE
TSK-1951: Extend WorkbasketPriorityReport for customInt

### PR DESCRIPTION
**Pull request by ryzheboka**
https://sonarcloud.io/summary/new_code?id=ryzheboka_taskana&branch=TSK-1951

---
Release Notes:

```
TSK-1951: add support for custom-int fields in reports
```
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [x] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [x] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] After integration of the PR, I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [ ] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [ ] Commit message format → TSK-XXX: Your commit message.
- [x] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

File: lib/taskana-core/src/main/java/pro/taskana/monitor/api/reports/TimeIntervalReportBuilder.java [hunk 1]
```diff
@@ -127,6 +129,16 @@ B customAttributeIn(TaskCustomField customField, String... strings)
...
 
+  B customIntAttributeIn(TaskCustomIntField customIntField, Integer... values)
+      throws InvalidArgumentException;
+
+  B customIntAttributeNotIn(TaskCustomIntField customIntField, Integer... values)
+      throws InvalidArgumentException;
+
+  B customIntAttributeWithin(TaskCustomIntField customIntField, IntInterval... values);
+
+  B customIntAttributeNotWithin(TaskCustomIntField customIntField, IntInterval... values);
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/main/java/pro/taskana/monitor/api/reports/TimeIntervalReportBuilder.java:140]: JavaDoc is missing

File: lib/taskana-core/src/test/java/acceptance/report/ProvideWorkbasketPriorityReportAccTest.java [hunk 2]
```diff
@@ -329,4 +336,173 @@ void should_ReturnItemsOfWorkbasketReport_When_FilteringWithAllCustomFieldsNotIn
...
+                  .withColumnHeaders(DEFAULT_TEST_HEADERS)
+                  .customIntAttributeIn(p.getLeft(), p.getRight())
+                  .inWorkingDays()
+                  .buildReport();
+          assertThat(report).isNotNull();
+          assertThat(report.rowSize()).isEqualTo(5);
+          assertThat(report.getRow("USER-1-1").getCells()).isEqualTo(new int[] {20, 0, 0});
+        };
+
+    return DynamicTest.stream(testCases.iterator(), p -> p.getLeft().name(), test);
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/report/ProvideWorkbasketPriorityReportAccTest.java:368]:
```suggestion
    return DynamicTest.stream(testCases.iterator(), Pair::getLeft, test);
```

I am not sure, does this work?

File: lib/taskana-core/src/test/java/acceptance/report/ProvideWorkbasketPriorityReportAccTest.java [hunk 3]
```diff
@@ -329,4 +336,173 @@ void should_ReturnItemsOfWorkbasketReport_When_FilteringWithAllCustomFieldsNotIn
...
+        p -> {
+          WorkbasketPriorityReport report =
+              MONITOR_SERVICE
+                  .createWorkbasketPriorityReportBuilder()
+                  .withColumnHeaders(DEFAULT_TEST_HEADERS)
+                  .customIntAttributeNotIn(p.getLeft(), p.getRight())
+                  .inWorkingDays()
+                  .buildReport();
+          assertThat(report).isNotNull();
+          assertThat(report.rowSize()).isEqualTo(0);
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/report/ProvideWorkbasketPriorityReportAccTest.java:395]:
```suggestion
          assertThat(report.rowSize()).isZero();
```

**Review by mustaphazorgati — CHANGES_REQUESTED** [review]

File: lib/taskana-core/src/test/java/acceptance/report/ProvideWorkbasketPriorityReportAccTest.java (hunk 2 above)
Comment (ryzheboka) [reply lib/taskana-core/src/test/java/acceptance/report/ProvideWorkbasketPriorityReportAccTest.java:368]: No, it doesn't compile

File: lib/taskana-core/src/test/java/acceptance/report/ProvideWorkbasketPriorityReportAccTest.java (hunk 3 above)
Comment (ryzheboka) [reply lib/taskana-core/src/test/java/acceptance/report/ProvideWorkbasketPriorityReportAccTest.java:395]: Ok :)

File: lib/taskana-core/src/main/java/pro/taskana/monitor/api/reports/TimeIntervalReportBuilder.java (hunk 1 above)
Comment (ryzheboka) [reply lib/taskana-core/src/main/java/pro/taskana/monitor/api/reports/TimeIntervalReportBuilder.java:140]: Ok, I will add it

**Review by mustaphazorgati — APPROVED** [review]

**Pull request merged**